core/rawdb/accessors_chain.go

```diff
@@ -337,6 +337,12 @@ func ReadHeader(db ethdb.Reader, hash common.Hash, number uint64) *types.Header
 		log.Error("Invalid block header RLP", "hash", hash, "err", err)
 		return nil
 	}
+	if header.IsEuclid && !header.EuclidHandled {
+		header.BlockSignature = header.Extra
+		header.Extra = make([]byte, len(header.Extra))
```

Comment:
This line will create a byte slice with length `header.Extra` where each byte will be 0. This is not equal to an empty slice or nil and we need to check how this behaves in RLP. My guess is that it would just treat it as any other normal byte slice and serialize it. Therefore, I think it should be 
```suggestion
		header.Extra = make([]byte, 0)
```
or 
```suggestion
		header.Extra = nil
```

Reply:
Yes. The final solution does `header.Extra. = []byte{}` as `header.Extra = nil` seems to break other parts of the codebase (not modified by me).